eventbus/kafka/eventsource/source_conn.go

```diff
@@ -0,0 +1,46 @@
+package eventsource
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+	"github.com/argoproj/argo-events/eventbus/common"
+	"github.com/argoproj/argo-events/eventbus/kafka/base"
+	"go.uber.org/zap"
+)
+
+type KafkaSourceConnection struct {
+	*base.KafkaConnection
+	Topic    string
+	Client   sarama.Client
+	Producer sarama.SyncProducer
+}
+
+func (c *KafkaSourceConnection) Publish(ctx context.Context, msg common.Message) error {
+	key := base.EventKey(msg.EventSourceName, msg.EventName)
```

Comment:
Hmm... the eventbus is a namespace scoped object, so on first thought including namespace in the key seems unnecessary, but it would be possible to create two eventbuses and point them at the same topic. Let me see if we have access to the namespace name here...

Reply:
I'm going to leave this as is. If two eventsources had the same name across two different namespaces **and** used the same topic then there would be much bigger fish to fry - sensors could potentially consume messages from the eventsource in another namespace if the event names were also the same.

What we really should do is guard against this altogether. If an eventsource already exists that is wired up to topic "x" and someone else creates another eventbus wired up to topic "x" we should probably reject creation of the second eventbus. I feel this can probably be a followup PR.